src/common.speech/SpeechServiceRecognizer.ts

```diff
@@ -92,28 +92,12 @@ export class SpeechServiceRecognizer extends ServiceRecognizerBase {
                 if (ResultReason.Canceled === resultReason) {
                     const cancelReason: CancellationReason = EnumTranslation.implTranslateCancelResult(simple.RecognitionStatus);
 
-                    result = new SpeechRecognitionResult(
-                        this.privRequestSession.requestId,
-                        resultReason,
+                    this.cancelRecognitionLocal(
```

Comment:
>cancelRecognitionLocal [](start = 25, length = 22)

cancelRecognitinLocal is an interesting name. It is a kind of wrapper for the real impl in speech, translator and intent.

It seems you are calling stopRecognizing multiple times. Since JS is single threaded, i assume you are fine.